bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -1158,7 +1158,10 @@ private static boolean checkIsRecordType(BRecordType sourceRecordType, BRecordTy
             Field sourceField = sourceFields.get(targetFieldEntry.getKey());
 
             if (sourceField == null) {
-                return false;
+                if (!SymbolFlags.isFlagOn(targetField.getFlags(), SymbolFlags.OPTIONAL)) {
```

Comment:
Shall we combine both `if` checks into one ?

Reply:
I will look into this.

Reply:
I cannot do this because there is a continue statement in the second if's else part.